### DISCUSSION TITLE
Sonicboom not perfectly working under tmux?

### DISCUSSION THREAD

**Discussion by joaociocca** [JanDeDobbeleer/oh-my-posh#2703] [Troubleshoot]
I'm not sure if commenting here would be the best place, but I'm not even sure where to ask. 

I just noticed that on Kali, after I set up OMP and sonicboom light, when I enter tmux I lose the error code return indicator. The following image is a capture from a horizontal split terminal, with the upper one being straight on zsh, and the lower one being on tmux:

![image](https://user-images.githubusercontent.com/6473725/186463117-aaded605-e8ab-45a3-9543-4779275c0078.png)

Any ideas what/how I could troubleshoot to fix this?

**Comment by JanDeDobbeleer** [comment]
@joaociocca no idea. Could it be tmux and multiline prompts are an issue?

**Reply by joaociocca** [reply]
I remember it working on Kali + TMUX before, around May when I was digging into OffSec's PWK and using Kali more. Had to do something with it today, and when I wrapped up my setup script, upon entering TMUX I noticed this...

I then retraced my steps manually, install brew, install OMP, get sonicboom's json... and the same thing happened. No idea what/where to check what could be wrong here. I don't think it's merely a tmux multiline prompt issue because in this capture, upper is plain pure zsh (disabled the OMP start command) and lower is tmux with pure zsh (also disabled OMP). No issue there, see?

![image](https://user-images.githubusercontent.com/6473725/186477592-17e7400d-90a7-4ced-870f-a69271878e99.png)

**Reply by JanDeDobbeleer** [reply]
@joaociocca I can't reproduce this in a clean setup on my end

**Reply by joaociocca** [reply]
just tried out with a freshly downloaded Kali 2022.3 official vbox file.

```bash
## install brew
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
echo 'eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"' >> /home/kali/.zshrc
eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"
sudo apt-get install build-essential
export PATH=/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:~/.local/bin:$PATH

## install oh-my-posh
brew install jandedobbeleer/oh-my-posh/oh-my-posh
oh-my-posh font install meslo
cur_shell="$(echo "$SHELL" | rev | awk -F'/' '{print $1}' | rev)"
if ! [ -f ~/sonicboom_light.omp.json ]; then 
    wget https://github.com/JanDeDobbeleer/oh-my-posh/raw/main/themes/sonicboom_light.omp.json -O sonicboom_light.omp.json
fi
eval "$(oh-my-posh init "$cur_shell" --config ~/sonicboom_light.omp.json)"
```

Then either restart term or open a new one to use installed font, and ran tmux:
![image](https://user-images.githubusercontent.com/6473725/187265912-4b080eb6-89ad-4e9c-8520-1855adf2f699.png)

**Reply by JanDeDobbeleer** [reply]
@joaociocca weird because I can't reproduce this on mac with tmux. Don't have the ability to run WSL2 on an M1 unfortunately.

**Reply by joaociocca** [reply]
any tips on what I could check? Because it's not just that the symbol is missing, sometimes pasting stuff or up/down history browse will mess with it even more (visually only)

**Reply by JanDeDobbeleer** [reply]
@joaociocca the thing is, oh-my-posh will output that line. It has no logic that would tell it otherwise which means the shell is somehow messing with it. If plain zsh displays the prompt correctly, you'll need to look at tmux (or the combination of Windows Terminal).

**Reply by joaociocca** [reply]
Ok, you gave me one idea I haven't tested yet: Windows Terminal remote into the Kali VM and check it out.

and... yeah, it's something with tmux, def.
![image](https://user-images.githubusercontent.com/6473725/187326787-0b2e3ce8-bee1-4135-8143-4aa400e96be7.png)

outside of tmux... it's all fine and nice.
![image](https://user-images.githubusercontent.com/6473725/187326944-ef164346-ed20-4915-aea6-1bcd9bde6940.png)

And then I thought "oh hey I have an ubuntu running elsewhere, and I never seen that happend there"... and yep, no problem on Ubuntu's tmux.
![image](https://user-images.githubusercontent.com/6473725/187327958-eedc0ae4-3965-4ada-8464-300d2beacce5.png)

Kali's tmux is 3.3a, Ubuntu's 3.0a

**Comment by joaociocca** [comment]
taking the dust off from here... I took a couple minutes to try other themes. One I found out has the same "color coded result" feature was powerlevel10k_classic. And that one works on Kali's tmux.
On the exact same panel, we can see here the Powerlevel working x Sonicboom failing.

![image](https://user-images.githubusercontent.com/6473725/199245348-5113d2f7-1280-4efc-b53d-a1c1c8428c8f.png)

I'll try to take some more time later to look into the similarities and differences between them to see if I find out and submit a PR for sonicboom. The thing I like most about it is that it keeps the timestamp and the result... while others clear everything away. I can understand the appeal for those who prefer it (like below, using slimfat), but up there comparing to Powerlevel10k, it's clear a bug on Sonicboom's configs.

![image](https://user-images.githubusercontent.com/6473725/199246812-b967d8f6-6d11-4af8-adda-4fd7aae18736.png)

(edit, inclusing below)

gets weirder! it seem that Sonicboom even keeps the first letter of the last command!
![image](https://user-images.githubusercontent.com/6473725/199247649-6972b9a3-bb9c-4264-8641-f0cb4e4b3b0d.png)

And here, on the "pure shell" on VSCode, it works nicely.
![image](https://user-images.githubusercontent.com/6473725/199247820-625060c3-ade9-4794-92a6-323556e8535d.png)

**Reply by joaociocca** [reply]
it seems that tmux may not like `\ufb0c` character. When I replace `"template": "\ufb0c ",` for Tokyo's `"template": "\u2514\u2500Δ",` it just works.
![image](https://user-images.githubusercontent.com/6473725/199256886-37ec1a27-a420-4cfa-bc65-f9674a906458.png)

Checking back and forth, and def seems to be something with tmux + `\ufb0c`. VScode is correctly configured to use MesloLGLDZ NF and shows the red/green arrows on pure shell, but when I activate Sonicboom on tmux, there's nothing.
![image](https://user-images.githubusercontent.com/6473725/199260868-7091acb8-933d-464d-8d7f-4af992302432.png)

If I edit Tokyo's config to use the same character, it's return is also gone.
![image](https://user-images.githubusercontent.com/6473725/199261235-fbe49a13-ecf1-4934-9cf7-e3ccf82a280d.png)

**Reply by joaociocca** [reply]
thought I'd try on linux VM to be absolutely sure. And there it is, Tmux dislikes it. (terminal is split near the 1 marker)
![image](https://user-images.githubusercontent.com/6473725/199264868-2f6cc035-b9f0-47a7-aa8b-a0fb05245256.png)

A simple echo of it will show on pure shell (zsh in this case), but won't show inside tmux. And now for a new one: it seems there can actually be something of a difference between Windows and Linux exhibition, because an arrow like `\uea9c` will show on Linux, even inside tmux, but not on Windows, even in pure shell!

(changing to \uea9c on Windows)
![image](https://user-images.githubusercontent.com/6473725/199266033-d1db3e99-5065-4aab-a582-5967ccab96c2.png)

(changing to \uea9c on Linux)
![image](https://user-images.githubusercontent.com/6473725/199266177-7d8b206d-2771-4a15-9d32-a60277b97a19.png)

**Reply by joaociocca** [reply]
And then I found that the Windows' culprit is the Meslo font. When changing to DaddyTimeMono NFM, the `uea9c` characters show up as expected, both inside tmux as outside.
![image](https://user-images.githubusercontent.com/6473725/199267940-9a0cf5d8-56a9-4fa8-a1ca-660f9544f41a.png)

But it still won't like `\ufb0c`. 
![image](https://user-images.githubusercontent.com/6473725/199268328-ed193e6d-2cd2-48d0-9f30-40b7cdf0138f.png)

**Reply by JanDeDobbeleer** [reply]
@joaociocca crazy find. It should still just handle that character.
